appIcons.js

```diff
@@ -166,7 +166,13 @@ const MyAppIcon = new Lang.Class({
 
     _updateRunningStyle: function() {
         this.parent();
```

Comment:
Hi again,

I went ahead and analyzed a bit more the signal handling... Here is a patch [f744dd66b5fdf04653a5d58da916483014b74a22] to better address this issue. I think it is more organized this way (and I now handle signals properly...).

Effectively, I moved `Shell.AppSystem.get_default().emit('installed-changed');` to docking.js, so that it has a proper initialization, and disconnection.